src/controllers/dishController.js

```diff
@@ -38,3 +38,20 @@ exports.getAllDishes = async (req, res) => {
     res.status(500).json({ error: 'Internal server error' });
   }
 };
+
+exports.getDishById = async (req, res) => {
+  const dishId = req.body.id;
+
+  try {
+    const dish = await Dish.findById(dishId);
+
+    if (!dish) {
+      res.status(404).json({ error: 'Dish not found' });
+    }
+
+    res.json(dish);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ error: 'Server error' });
+  }
+};
```

Comment:
This controller will not work, because you are using `const dishId = req.body.id` The correct way to do that by using the params from from request id like this `const dishId = req.params.id`

Reply:
actually the both commands work for me
but for testing on postman, if we use const dishId = req.body.id you should choice body then raw and write this
 {
 "id": "  "  
}  

if we use const dishId = req.params.id we should choice params on postman when testing